mtbf_job_runner.py

```diff
@@ -110,28 +108,16 @@ def collect_memory_report(self):
 
     def get_free_device(self):
         dp = DevicePool(self.serial)
-        if dp and dp.get_lock():
+        self.dp = dp
+        do = dp.get_device()
+        if dp and do:
```

Comment:
dp should never get None as return value...line 110 will only init an device pool object...
and the line 112 will possible to return None if we do not have any device could use...